### PR TITLE
Ensure the default length calculation includes the content type byte

### PR DESCRIPTION
**Pull request by mattcaswell**
TLSv1.3 includes an extra byte after the payload for the content type.
We should incorporate that in the calculation of the default buffer length.

We already had a test capable of detecting this, but this failure only occurs if SSL3_ALIGN_PAYLOAD has been set to a non default value (i.e. 4) - which never occurs in our CI. I've now added such a check to our daily runchecker builds.

The last commit should be dropped before merge. It is there only to make the newly added CI job run in this PR.

Fixes https://github.com/openssl/openssl/issues/23015

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Comment by mattcaswell** [comment]
Ping @openssl/committers for second review

**Review by hlandau — APPROVED** [review]

**Comment by mattcaswell** [comment]
Pushed to master/3.2. Thanks.

**Pull request closed without merging**
